File: src/main/java/com/autobook/Social/Post/Post.java
package com.autobook.Social.Post;

import jakarta.persistence.*;
import com.autobook.Enum.PostType;
import com.autobook.Social.User.User;
import com.autobook.Social.Comment.Comment;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "posts")
@Getter
@Setter
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "content", columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(name = "image_url")
    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @Enumerated(EnumType.STRING)
    @Column(name = "post_type", nullable = false, length = 20)
    private PostType postType;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "like_count", nullable = false)
    private int likeCount = 0;

    @Column(name = "comment_count", nullable = false)
    private int commentCount = 0;

    @Column(name = "repost_count", nullable = false)
    private int repostCount = 0;

    @OneToMany(mappedBy = "post", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments = new ArrayList<>();
}

File: frontend/src/domain/models/Post.ts
import type { UserCard } from './User';

// Post

export interface Post {
  id: number;
  content: string;
  author: UserCard;
  postType: 'FEED' | 'REVIEW' | 'ANNOUNCEMENT';
  imageUrl: string | null;
  hasImage: boolean;
  createdAt: string;
  updatedAt: string;
  likeCount: number;
  commentCount: number;
  repostCount: number;
  likedByMe: boolean;
  repostedByMe: boolean;
}

export interface PostDetails extends Post {
  comments: Comment[];
}

export interface CreatePostRequest {
  content: string;
  postType: string;
  imageUrl: string | null;
}

// Comment

export interface Comment {
  id: number;
  content: string;
  author: UserCard;
  createdAt: string;
  updatedAt: string;
  parentId: number | null;
  replies: Comment[];
  likes: number;
}

export interface CreateCommentRequest {
  content: string;
  parentId: number | null;
}
